crates/c-api/doxygen.conf

```diff
@@ -0,0 +1,2538 @@
+# Doxyfile 1.8.17
```

Comment:
It's worth pointing out that I generated this file with `doxygen -f doxygen.conf`. I've tweaked a number of settings internally but I'm hoping that most of this file can basically be ignored.